showthread.php

```diff
@@ -734,15 +734,32 @@
 	}
 
 	// Increment the thread view.
-	if($mybb->settings['delayedthreadviews'] == 1)
+	$count_view = true; // By default, we count the view. Only if one of the below conditions is met do we not count it.
+	if($mybb->settings['threadviewcountexcludespiders'] == 1 && $session->is_spider == true)
 	{
-		$db->shutdown_query("INSERT INTO ".TABLE_PREFIX."threadviews (tid) VALUES('{$tid}')");
+		$count_view = false;
 	}
-	else
+	if($mybb->settings['threadviewcountexcludeguests'] == 1 && $mybb->user['uid'] == 0)
+	{
+		$count_view = false;
+	}
+	if($mybb->settings['threadviewcountexcludethreadauthor'] == 1 && $mybb->user['uid'] == $thread['uid'])
+	{
+		$count_view = false;
+	}
+	
+	if($count_view == true)
```

Comment:
It might be better to use a single `if` condition for the block below

Reply:
Yes, you're right, it might be. @Eldenroot and I were going for an "Easy enough for a total noob to grok" control flow.

Reply:
@dvz - Done, simplified!